external-testing/README.md

```diff
@@ -0,0 +1,138 @@
+* [2] https://requre.readthedocs.io/en/latest/usages/postprocessing.html
+* [3] [You can use a decorator or a context manager](https://vcrpy.readthedocs.io/en/latest/usage.html)
+* [4] [Tenths of lines of code](https://github.com/packit-service/packit/blob/master/tests_recording/__init__.py#L16) [which are very hard to understand](https://github.com/packit-service/packit/blob/master/tests_recording/testbase.py#L22)
+* [5] Betamax has [pluggable matching mechanism](https://betamax.readthedocs.io/en/latest/matchers.html)
+* [6] This is by design: https://github.com/betamaxpy/betamax/issues/125#issuecomment-267405773
+* [7] https://betamax.readthedocs.io/en/latest/serializers.html
+
+### Feelings
+
+These are my personal (@TomasTomecek) comments and opinions:
+* VCRpy's API and documentation is focused on the record & replay use case
+  which makes it very pleasant to work with.
+* requre's documentation and API is centered around implementation details
+  which makes it very configurable, but unfortunately extremely hard to work
+  with - even to the point that you need to understand internals.
+* VCRpy in action:
+  * [Basic usage](https://github.com/packit-service/packit/pull/785)
+  * [Calling the same URL twice](https://github.com/packit-service/ogr/pull/373)
+* The fact that requre is matching replay via call stack makes it less flexible
+  and requires data regeneration more often than VCRpy.
+* In my opinion: VCRpy masters its job very well.
+
+
+## So, what's next?
+
+Hunor asked me to do an in-depth analysis for the possible solutions. We need
+a team discussion to pick one.
+
+
+### Solution 1: Keep using requre and improve it
+
+We can keep using requre as we are now. But given that VCRpy has
+* very neat user experience
+* and matches requests based on their data and not the call stack,
+
+it is very compelling to have VCRpy's experience. This means that if we stick
+to using requre, we need to improve it to match VCRpy's UX.
+
+21+ story points probably and weeks of polishing it (2 sprints at least)
+
+
+### Solution 2: Switch to VCRpy
+
+Alternatively, we can switch to VCR. There are multiple sub-options.
+
+
+#### a) Get requre's features in there
+
+Since VCRpy has "dead upstream" we would either ended up
+* maintaining the project
+* or forking it
+
+Definitely more work than implementing things in requre itself.
+
+21++ story points (2-3 sprints)
+
+
+#### b) VCRpy + flexmock + pytest fixtures
+
+This implies ditching requre completely. We'd need to:
+* rewrite existing tests to use VCR instead of requre (easy)
+* regenerate or transform data (should be easy)
+* use flexmock (or something else) for git, fedpkg...
+* most importantly: prepare the structure well so it's not messy
+
+This should be achievable in 2 sprints.
+
+Technical arguments end here and personal preference would follow.
+
+
+### Not a solution 3: VCRpy + requre
+
+They don't play well together. This solution doesn't make any sense.
```

Comment:
Have you tried also avoiding requre touching the `requests`? I am not sure how deeply you were trying that...